src/db/queries.ts

```diff
@@ -1,103 +1,251 @@
-        throw error
-    }
+async function getUserByIdTypeORM(token: string) {
+    const user = await createQueryBuilder("user")
+        .select('user')
+        .from(User, 'user')
+        .where("user.token = :token", { token: token })
+        .leftJoinAndSelect("user.searches", "search")
+        .getOne();
+
+    // const user = await AppDataSource
+    // .getRepository(User)
+    // .createQueryBuilder("user")
+    // .leftJoinAndSelect("user.searches", "search")
+    // .where("user.token = :token", { token: token })
+    // .getOne()
+    return user;
 }
 
-//update user info
-const updateUser = async (token: string, lang: string) => {
-    const user = await pool.query('UPDATE users SET lang = $1 WHERE id = $2', [lang, token]);
-    console.log(user.rows);
+// //update user function
+async function updateUserTypeORM(token: string, lang: string) {
+    const user = User.create({
+        lang: lang,
+        searches: []
+    });
+    await user.save();
+    return user;
+
+    //     const userRepository = AppDataSource.getRepository(User)
+    // const userToUpdate = await userRepository.findOneBy({
+    //     token: token,
+    // })
+    // userToUpdate!.lang = lang;
+    // await userRepository.save(userToUpdate!);
 }
 
-//delete user
-const deleteUser = (token: string) => {
-    pool.query('DELETE FROM users WHERE id = $1', [token], (error: unknown, results: unknown) => {
-        if (error) {
-            throw error
-        } else {
-            console.log(results);
-            return (`User deleted with ID: ${token}`)
-        }
-    })
+//remove user function
+async function removeUserTypeORM(token: string) {
+    const res = await User.delete(token);
+    return res
+    // const userRepository = AppDataSource.getRepository(User)
+    // const userToUpdate = await userRepository.findOneBy({
+    //     token: token,
+    // })
+    // await userRepository.remove(userToUpdate!);
 }
 
-//export
-module.exports = {
-    getUsers,
-    getUserById,
-    createUser,
-    updateUser,
-    deleteUser,
+//add search
+async function addSearchTypeORM(user: User, title: string) {
+    const search = Search.create({
+        searchTitle: title,
+        searchTime: new Date().getDate()
+    });
+    user.addSearch(search);
+    console.log(user);
+    user.save();
+
+    // search.searchTitle = title;
+    // search.searchTime = new Date().getDate();
+    // user.addSearch(search);
+    // const userRepository = AppDataSource.getRepository(User);
+    // await userRepository.save(user!);
+    // await getAllUsersTypeORM();
 }
 
 
-// BL VS CONTROLLER - need to seprate the two. all error handle in next()
-//config file by enviorment variables. conifg.ts - > file.env (gen variables & secrets) not in commit.
-// https://www.npmjs.com/package/dotenv
-// use const or let not var. immutable concept.
-//unknow is better then any => typeof === blah
-//use next(e)
-//add try catch in async
\ No newline at end of file
+module.exports = {
```

Comment:
You should either use `import / export` or `require / module.exports` in 1 project